chain-core/src/init/config.rs

```diff
@@ -72,12 +73,32 @@ impl SlashRatio {
         }
     }
 
+    pub fn get_proportional(
+        self,
+        failing_validators: usize,
+        total_validators: usize,
+    ) -> SlashRatio {
+        let millis =
+            (((3.0 * failing_validators as f64) / total_validators as f64).powi(2) * 1000.0) as u64;
+        SlashRatio::new(Milli::from_millis(std::cmp::min(millis, 1000))).unwrap() * self
```

Comment:
this should use / preserve fixed-point operations rather than casting to floating points, otherwise it may not be deterministic (e.g. different rustc/llvm versions may optimizing floating point operations differently).
`x.powi(2)` can be rewritten as `x*x`